<robot name="vs087_1_description" xmlns:xacro="http://ros.org/wiki/xacro">

<xacro:macro name="vs087_1" params="parent origin_x origin_y origin_z roll pitch yaw">
	
	<joint name="joint_w_1" type="fixed">
        <origin xyz="${origin_x} ${origin_y} ${origin_z}" rpy="${roll} ${pitch} ${yaw}"/>
        <parent link="${parent}"/>
		<child link="base_link_1"/>
	</joint>
	<link name="base_link_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/base_link.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/base_link.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<link name="J1_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J1.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J1.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_1_1" type="revolute">
		<parent link="base_link_1"/>
		<child link="J1_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.197500"/>
		<axis xyz="-0.000000 -0.000000 1.000000"/>
		<limit effort="1" lower="-2.96705972839036" upper="2.96705972839036" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_1_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_1_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_1_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<link name="J2_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J2.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J2.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_2_1" type="revolute">
		<parent link="J1_1"/>
		<child link="J2_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="0.030000 0.000000 0.197500"/>
		<axis xyz="-0.000000 1.000000 -0.000000"/>
		<limit effort="1" lower="-1.74532925199433" upper="2.35619449019234" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_2_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_2_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_2_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<link name="J3_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J3.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J3.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_3_1" type="revolute">
		<parent link="J2_1"/>
		<child link="J3_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.445000"/>
		<axis xyz="-0.000000 1.000000 -0.000000"/>
		<limit effort="1" lower="-2.37364778271229" upper="2.67035375555132" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_3_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_3_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_3_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<link name="J4_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J4.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J4.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_4_1" type="revolute">
		<parent link="J3_1"/>
		<child link="J4_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="-0.020000 0.000000 0.287000"/>
		<axis xyz="-0.000000 -0.000000 1.000000"/>
		<limit effort="1" lower="-4.71238898038469" upper="4.71238898038469" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_4_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_4_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_4_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<link name="J5_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J5.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J5.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_5_1" type="revolute">
		<parent link="J4_1"/>
		<child link="J5_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.143000"/>
		<axis xyz="-0.000000 1.000000 -0.000000"/>
		<limit effort="1" lower="-2.0943951023932" upper="2.0943951023932" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_5_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_5_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_5_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>
	<link name="J6_1">
		<visual>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J6.dae" scale="1 1 1"/>
			</geometry>
		</visual>
		<collision>
			<origin rpy="0 0 0" xyz="0 0 0"/>
			<geometry>
				<mesh filename="package://denso_robot_descriptions/vs087_description/J6.dae" scale="1 1 1"/>
			</geometry>
		</collision>
		<inertial>
			<mass value="1"/>
			<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.000000"/>
			<inertia ixx="1" ixy="0" ixz="0" iyy="1" iyz="0" izz="1"/>
		</inertial>
	</link>
	<joint name="joint_6_1" type="revolute">
		<parent link="J5_1"/>
		<child link="J6_1"/>
		<origin rpy="0.000000 0.000000 0.000000" xyz="0.000000 0.000000 0.080000"/>
		<axis xyz="-0.000000 -0.000000 1.000000"/>
		<limit effort="1" lower="-6.28318530717959" upper="6.28318530717959" velocity="1"/>
		<dynamics damping="0" friction="0"/>
	</joint>
	<transmission name="trans_6_1">
		<type>transmission_interface/SimpleTransmission</type>
		<joint name="joint_6_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
		</joint>
		<actuator name="motor_6_1">
			<hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
			<mechanicalReduction>1</mechanicalReduction>
		</actuator>
	</transmission>

    </xacro:macro>
</robot>
